torch/csrc/cuda/python_comm.cpp

```diff
@@ -63,6 +92,17 @@ void initCommMethods(PyObject *module) {
           py::arg("tensors"),
           py::arg("dim"),
           py::arg("destination_index"),
+          py::call_guard<py::gil_scoped_release>())
+      .def(
+          "_gather_out",
```

Comment:
This is prior to this PR. Just curious, why we don't support providing streams for gather as well?

Reply:
:) I don't know. I assume that scatter was specially handled to speed up DP.